uw-frame-components/portal/messages/controllers.js

```diff
@@ -0,0 +1,590 @@
+                seenAnnouncementIds.push($scope.latestAnnouncement.id);
+                messagesService.setMessagesSeen(seenAnnouncementIds);
+              });
+            };
+            displayPopup();
+          }
+        };
+
+        /**
+         * Set the mascot image and its fallback
+         */
+        var setMascot = function() {
+          if ($rootScope.portal && $rootScope.portal.theme) {
+            vm.mascotImage =
+              $rootScope.portal.theme.mascotImg || 'img/robot-taco.gif';
+          } else {
+            vm.mascotImage = 'img/robot-taco.gif';
+          }
+          // https://github.com/Gillespie59/eslint-plugin-angular/issues/231
+          // eslint-disable-next-line angular/on-watch
+          $rootScope.$watch('portal.theme', function(newVal, oldVal) {
+            if (newVal !== oldVal) {
+              vm.mascotImage = newVal.mascotImg || 'img/robot-taco.gif';
+            }
+          });
+        };
+
+        // ////////////////
+        // Scope methods //
+        // ////////////////
+        /**
+         * Remove dismissed announcement from scope announcements,
+         * then update storage
+         * @param id
+         */
+        vm.markSingleAnnouncementSeen = function(id) {
+          // Use $filter to filter out by ID
+          vm.announcements = $filter('filterOutMessageWithId')(
+            vm.announcements,
+            id
+          );
+          // Add to seenAnnouncementsIds
+          seenAnnouncementIds.push(id);
+          // Call service to save results
+          messagesService.setMessagesSeen(seenAnnouncementIds);
+        };
+
+        /**
+         * Add all IDs of unseen announcements to the seenAnnouncements
+         * array, then call the messagesService to save results
+         */
+        vm.markAllAnnouncementsSeen = function() {
+          angular.forEach(separatedAnnouncements.unseen, function(value) {
+            seenAnnouncementIds.push(value.id);
+          });
+          messagesService.setMessagesSeen(seenAnnouncementIds);
+        };
+
+        /**
+         * Make mascot bobble when hovered
+         */
+        vm.toggleHover = function() {
+          vm.hover = vm.hover ? false : true;
+        };
+
+        /**
+         * Make mascot appear when clicked
+         */
+        vm.toggleActive = function() {
+          vm.active = !vm.active;
+        };
+
+        /**
+         * Reset mascot when menu is closed
+         */
+        $scope.$on('$mdMenuClose', function() {
+          vm.hover = false;
+          vm.active = false;
+        });
+      }])
+
+    .controller('FeaturesPageController', ['$log', '$filter',
+      'messagesService', 'MESSAGES', 'MISC_URLS',
+      function($log, $filter, messagesService, MESSAGES, MISC_URLS) {
+        var vm = this;
+
+        vm.announcements = [];
+        vm.MISC_URLS = MISC_URLS;
+
+        if (MESSAGES.announcementsEnabled) {
+          // Check if group filtering is enabled
```

Comment:
Is there implementation missing that this comment was meant to be commenting upon?